transform-plugins/src/main/java/co/cask/hydrator/plugin/Encoder.java

```diff
@@ -95,6 +95,23 @@ private void parseConfiguration(String config) throws IllegalArgumentException {
   public void configurePipeline(PipelineConfigurer pipelineConfigurer) throws IllegalArgumentException {
     super.configurePipeline(pipelineConfigurer);
     parseConfiguration(config.encode);
+
+    Schema inputSchema = pipelineConfigurer.getStageConfigurer().getInputSchema();
+    // for the fields in input schema, if they are to be encoded (if present in encodeMap)
+    // make sure their type is either String or Bytes and throw exception otherwise
+    if (inputSchema != null) {
+      for (Schema.Field field : inputSchema.getFields()) {
+        if (encodeMap.containsKey(field.getName())) {
+          if (!field.getSchema().getType().equals(Schema.Type.BYTES) &&
+            !field.getSchema().getType().equals(Schema.Type.STRING)) {
+            throw new IllegalArgumentException(
+              String.format("Field %s is of type %s in Input Schema, currently only String and Bytes can be encoded",
```

Comment:
use the same error message as above for consistency:

"Field %s must be of type bytes or string. It is currently of type %s"